### PR TITLE
[inductor] Freeze the layout of the conv input to channels_last

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122765

Fix https://github.com/pytorch/pytorch/issues/118082.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/graph.py [hunk 1]
```diff
@@ -1102,28 +1102,41 @@ def debug(msg):
...
                             result = ir.ExternKernel.require_stride_order(
                                 result, ir.get_stride_order(n.meta["val"].stride())
                             )
+                        if user.target in need_fixed_channels_last_layout:
+                            channels_last_stride_order = [0] + list(
+                                reversed(range(1, len(result.get_size()) - 1))
+                            )
+                            channels_last_stride_order = [
+                                len(channels_last_stride_order)
+                            ] + channels_last_stride_order
```
Comment (jgong5) [inline torch/_inductor/graph.py:1136]: can we use `make_channels_last_strides_for`?

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/_inductor/graph.py (hunk 1 above)
Comment (jiayisunx) [reply torch/_inductor/graph.py]: I have modified it, thanks for your suggestion.

**Review by jgong5 — APPROVED** [review]

**Review by leslie-fang-intel — APPROVED** [review]

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by jiayisunx** [comment]
@jansel , could you please review this PR? Thanks.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**